src/rulesets/Base/arraymath.jl

```diff
@@ -104,3 +104,14 @@ function rrule(::typeof(\), b::Real, A::AbstractArray{<:Real})
     end
     return Y, backslash_pullback
 end
+
+#####
+##### Negation (Unary -)
+#####
+
+function rrule(::typeof(-), x::AbstractArray)
+    function negation_pullback(ȳ)
+        return NO_FIELDS, InplaceableThunk(@thunk(-ȳ), ā -> ā .-= ȳ)
```

Comment:
This is so much nicer than what we had before with `_subtract`.

I am trying to understand `InplaceableThunk`s [`add!` function](https://github.com/JuliaDiff/ChainRulesCore.jl/blob/master/src/differentials/thunks.jl#L108-L109), where does it get called?

The comment about it needing to be done `more efficiently` is a bit confusing. Does it mean doing
`ā -> ā .-= ȳ` instead of the suggested`ā -> ā .+= -ȳ` in this case?

Reply:
`ā -> ā .+= -ȳ` will allocate a matrix to hold `-ȳ`
Allocation could be avoided via fusing .
`ā -> ā .+= .-ȳ`
but that is pretty confusing looking